### PR TITLE
etcdctl/ctlv3: make 'table' printer configurable

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5296.

### CODE REVIEW

**Comment by heyitsanthony** [comment]
@gyuho I was thinking this would have a different `-w` option for table mode versus non-table mode and two separate `printer` implementations (e.g., `simplePrinter` and `tablePrinter`) to go along with that.

**Comment by heyitsanthony** [comment]
I noticed my comment on #5296 said to use a separate flag but I somehow confused separate flag with a different format option. Sorry :(

**Comment by gyuho** [comment]
Np! Yeah you mentioned printer. I will fix it to use that. Thanks!

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Made others configurable with `-w table` flag to be consistent.

File: etcdctl/ctlv3/command/ep_command.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ func newEpStatusCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:   "status",
 		Short: "status prints out the status of endpoints specified in `--endpoints` flag",
+		Long:  "prints comma-separated status lists (endpoint, ID, version, db size, is leader, raft term, raft index)",
```
Comment (xiang90) [inline etcdctl/ctlv3/command/ep_command.go]: the long is not accurate. we might print out in other format.

**Comment by xiang90** [comment]
@gyuho We just need to change the tablewriter to use `,` as separator, see https://github.com/olekukonko/tablewriter/blob/master/table.go#L144-L151

**Comment by gyuho** [comment]
@xiang90 Fixed the `Long` description to clarify the use case for `simple` flag.

@heyitsanthony suggested to implement different printer (one for `simplePrinter` and one for `tablePrinter`). So when we do `simplePrinter`, do we also want to use `tablewriter`?

File: etcdctl/ctlv3/command/printer.go [hunk 2]
```diff
@@ -51,10 +51,57 @@ func NewPrinter(printerType string, isHex bool) printer {
 		return &jsonPrinter{}
 	case "protobuf":
 		return &pbPrinter{}
+	case "table":
+		return &tablePrinter{}
 	}
 	return nil
 }
 
+func getMemberListRows(r v3.MemberListResponse) (hdr []string, rows [][]string) {
```
Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: generateMemberListTable

File: etcdctl/ctlv3/command/printer.go [hunk 3]
```diff
@@ -51,10 +51,57 @@ func NewPrinter(printerType string, isHex bool) printer {
...
+			status,
+			m.Name,
+			strings.Join(m.PeerURLs, ","),
+			strings.Join(m.ClientURLs, ","),
+		})
+	}
+	return
+}
+
+func getEndpointStatusRows(statusList []epStatus) (hdr []string, rows [][]string) {
```
Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: makeEndpointStatusTable

File: etcdctl/ctlv3/command/printer.go [hunk 4]
```diff
@@ -51,10 +51,57 @@ func NewPrinter(printerType string, isHex bool) printer {
...
+			fmt.Sprint(humanize.Bytes(uint64(status.Resp.DbSize))),
+			fmt.Sprint(status.Resp.Leader == status.Resp.Header.MemberId),
+			fmt.Sprint(status.Resp.RaftTerm),
+			fmt.Sprint(status.Resp.RaftIndex),
+		})
+	}
+	return
+}
+
+func getDBStatusRows(ds dbstatus) (hdr []string, rows [][]string) {
```
Comment (xiang90) [inline etcdctl/ctlv3/command/printer.go]: makeDBstatusTable

File: etcdctl/ctlv3/command/ep_command.go [hunk 5]
```diff
@@ -51,6 +51,7 @@ func newEpStatusCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:   "status",
 		Short: "status prints out the status of endpoints specified in `--endpoints` flag",
+		Long:  "when 'write-out' is simple, prints comma-separated status lists (endpoint, ID, version, db size, is leader, raft term, raft index)",
```
Comment (xiang90) [inline etcdctl/ctlv3/command/ep_command.go]: when --write-out is set to simple, this command prints out comma-separated status lists for each endpoint.
The items in the lists are endpoint, ID, version, db size ...

**Comment by gyuho** [comment]
@xiang90 Just changed the method names to `make...Table`, and fixed the `Long` description.

PTAL.

Thanks!

File: etcdctl/ctlv3/command/member_command.go [hunk 6]
```diff
@@ -84,6 +84,7 @@ func NewMemberListCommand() *cobra.Command {
 	cc := &cobra.Command{
 		Use:   "list",
 		Short: "list is used to list all members in the cluster",
+		Long:  "when --write-out is set to simple, this command prints out comma-separated member lists for each endpoint. The items in the lists are ID, Status, Name, Peer Addrs, Client Addrs.",
```
Comment (xiang90) [inline etcdctl/ctlv3/command/member_command.go]: multiple lines? using ``` ```?

**Comment by gyuho** [comment]
@xiang90  Just made `Long` descriptions multiple-lines. PTAL.

Thanks!

**Comment by xiang90** [comment]
LGTM defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**